api/src/pcapi/core/bookings/api.py

```diff
@@ -645,18 +645,27 @@ def _cancel_bookings_from_stock(
     Cancel multiple bookings and update the users' credit information on Batch.
     Note that this will not reindex the stock.offer in Algolia
     """
-    deleted_bookings: list[Booking] = []
+    cancelled_bookings: list[Booking] = []
     for booking in stock.bookings:
+        # Do not make several SQL queries per booking then rollback to savepoint for those which cannot be cancelled.
+        # This optimization could avoid timeout in the backoffice when an EAN is rejected with many past bookings.
+        if booking.status in (BookingStatus.REIMBURSED, BookingStatus.CANCELLED):
+            continue
+
+        cancel_even_if_used = typing.cast(bool, stock.offer.isEvent)
```

Comment:
```suggestion
        cancel_even_if_used = bool(stock.offer.isEvent)
```
c'est marginalement plus lent mais je trouve ça plus intuitif

Reply:
J'ai juste déplacé l'expression existante, je ne sais pas pourquoi `typing.cast`  était utilisé.
D'autant plus que la propriété `offer.isEvent` renvoie déjà un `bool`.

Reply:
En version actuelle, ni mypy ni pylint, dans l ne se plaint si je simplifie sans cast.